### ISSUE TITLE
ESP does not connect to router after credential save. 

### ISSUE THREAD

**Issue by RudyFiero** [lubosmato/esp32-essentials#6]
I am able to connect to the ESP with it's AP. I can enter my router information. The serial output says the information was saved. 
It then looks like it does try to disconnect but it fails. I have other things connected to the router, I can see that it is working. The following is the ESP output. 

=================================
I left the output below. It is what I got. I don't know why. I then went and reset my router. then did a hardware reset to the ESP, then it did connect. I need to look into this in the morning.

=================================
`
ESP-ROM:esp32s2-rc4-20191025
Build:Oct 25 2019
rst:0x1 (POWERON),boot:0x8 (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
mode:DIO, clock div:1
load:0x3ffe6100,len:0x1828
load:0x4004c000,len:0x9e8
load:0x40050000,len:0x2b70
SHA-256 comparison failed:
Calculated: b4d99850fe459f149546ea9e76101766231b202d18566aeadf007958ecbb83ef
Expected: 53829e24d1b54c6c6f2f4888beb5a3b700837dc2c5a5f35b8fdc8ed825c89be5
Attempting to boot anyway...
entry 0x4004c1ec
[1B][0;32mI (40) boot: ESP-IDF v4.3-dirty 2nd stage bootloader[1B][0m
[1B][0;32mI (40) boot: compile time 15:49:06[1B][0m
[1B][0;32mI (40) boot: chip revision: 0[1B][0m
[1B][0;32mI (43) boot.esp32s2: SPI Speed      : 80MHz[1B][0m
[1B][0;32mI (47) boot.esp32s2: SPI Mode       : DIO[1B][0m
[1B][0;32mI (52) boot.esp32s2: SPI Flash Size : 4MB[1B][0m
[1B][0;32mI (57) boot: Enabling RNG early entropy source...[1B][0m
[1B][0;32mI (62) boot: Partition Table:[1B][0m
[1B][0;32mI (66) boot: ## Label            Usage          Type ST Offset   Length[1B][0m
[1B][0;32mI (73) boot:  0 nvs              WiFi data        01 02 00009000 00006000[1B][0m
[1B][0;32mI (81) boot:  1 phy_init         RF data          01 01 0000f000 00001000[1B][0m
[1B][0;32mI (88) boot:  2 factory          factory app      00 00 00010000 00100000[1B][0m
[1B][0;32mI (95) boot: End of partition table[1B][0m
[1B][0;32mI (100) esp_image: segment 0: paddr=00010020 vaddr=3f000020 size=1e688h (124552) map[1B][0m
[1B][0;32mI (135) esp_image: segment 1: paddr=0002e6b0 vaddr=3ffc6440 size=01968h (  6504) load[1B][0m
[1B][0;32mI (137) esp_image: segment 2: paddr=00030020 vaddr=40080020 size=7ebf0h (519152) map[1B][0m
[1B][0;32mI (252) esp_image: segment 3: paddr=000aec18 vaddr=3ffc7da8 size=024ech (  9452) load[1B][0m
[1B][0;32mI (255) esp_image: segment 4: paddr=000b110c vaddr=40022000 size=14438h ( 83000) load[1B][0m
[1B][0;32mI (280) esp_image: segment 5: paddr=000c554c vaddr=50000000 size=00010h (    16) load[1B][0m
[1B][0;32mI (291) boot: Loaded app from partition at offset 0x10000[1B][0m
[1B][0;32mI (291) boot: Disabling RNG early entropy source...[1B][0m
[1B][0;32mI (303) ca[0F][06]žf†~[00]˜[18]†ž[1E]FÉÕÑ¥½¹cache 	: size 8KB, 4Ways, cache line size 32Byte[1B][0m
[1B][0;32mI (303) cpu_start: Pro cpu up.[1B][0m
[1B][0;32mI (317) cpu_start: Pro cpu start user code[1B][0m
[1B][0;32mI (317) cpu_start: cpu freq: 160000000[1B][0m
[1B][0;32mI (317) cpu_start: Application information:[1B][0m
[1B][0;32mI (322) cpu_start: Project name:     blink[1B][0m
[1B][0;32mI (326) cpu_start: App version:      264977f-dirty[1B][0m
[1B][0;32mI (332) cpu_start: Compile time:     Aug  4 2021 15:49:15[1B][0m
[1B][0;32mI (338) cpu_start: ELF file SHA256:  d0844d00afc260c8...[1B][0m
[1B][0;32mI (344) cpu_start: ESP-IDF:          v4.3-dirty[1B][0m
[1B][0;32mI (349) heap_init: Initializing. RAM available for dynamic allocation:[1B][0m
[1B][0;32mI (356) heap_init: At 3FF9E000 len 00002000 (8 KiB): RTCRAM[1B][0m
[1B][0;32mI (363) heap_init: At 3FFCE9B0 len 0002D650 (181 KiB): DRAM[1B][0m
[1B][0;32mI (369) heap_init: At 3FFFC000 len 00003A10 (14 KiB): DRAM[1B][0m
[1B][0;32mI (375) spi_flash: detected chip: generic[1B][0m
[1B][0;32mI (380) spi_flash: flash io: dio[1B][0m
[1B][0;32mI (388) cpu_start: Starting scheduler on PRO CPU.[1B][0m
[1B][0;32mI (411) wifi: disconnecting wifi[1B][0m
[1B][0;31mE (411) wifi_init: Failed to deinit Wi-Fi driver (0x3001)[1B][0m
[1B][0;32mI (411) wifi: connecting to wifi[1B][0m
I (421) wifi:wifi driver task: 3ffd66c4, prio:23, stack:6656, core=0
[1B][0;32mI (421) system_api: Base MAC address is not set[1B][0m
[1B][0;32mI (421) system_api: read default base MAC address from EFUSE[1B][0m
I (441) wifi:wifi firmware version: c7d0450
I (441) wifi:wifi certification version: v7.0
I (441) wifi:config NVS flash: enabled
I (441) wifi:config nano formating: disabled
I (441) wifi:Init data frame dynamic rx buffer num: 32
I (451) wifi:Init management frame dynamic rx buffer num: 32
I (451) wifi:Init management short buffer num: 32
I (461) wifi:Init dynamic tx buffer num: 32
I (461) wifi:Init static rx buffer size: 1600
I (471) wifi:Init static rx buffer num: 10
I (471) wifi:Init dynamic rx buffer num: 32
[1B][0;32mI (471) wifi_init: rx ba win: 6[1B][0m
[1B][0;32mI (481) wifi_init: tcpip mbox: 32[1B][0m
[1B][0;32mI (481) wifi_init: udp mbox: 6[1B][0m
[1B][0;32mI (491) wifi_init: tcp mbox: 6[1B][0m
[1B][0;32mI (491) wifi_init: tcp tx win: 5744[1B][0m
[1B][0;32mI (491) wifi_init: tcp rx win: 5744[1B][0m
[1B][0;32mI (501) wifi_init: tcp mss: 1440[1B][0m
[1B][0;32mI (501) wifi_init: WiFi IRAM OP enabled[1B][0m
[1B][0;32mI (511) wifi_init: WiFi RX IRAM OP enabled[1B][0m
[1B][0;32mI (521) phy_init: phy_version 1800,e7ef680,Apr 13 2021,11:45:08[1B][0m
I (661) wifi:mode : sta (7c:df:a1:00:a6:1a)
I (661) wifi:enable tsf
[1B][0;32mI (661) wifi: trying to connect to AP SSID: 'HeatPad', password: '09876543210'...[1B][0m
Waiting for connection...
I (681) wifi:new:<1,0>, old:<1,0>, ap:<255,255>, sta:<1,0>, prof:5
I (681) wifi:state: init -> auth (b0)
Waiting for connection...
I (1681) wifi:state: auth -> init (200)
I (1681) wifi:new:<1,0>, old:<1,0>, ap:<255,255>, sta:<1,0>, prof:5
[1B][0;32mI (1681) wifi: re-trying to connect to the AP[1B][0m
Waiting for connection...
Waiting for connection...
[1B][0;32mI (3741) wifi: re-trying to connect to the AP[1B][0m
I (3751) wifi:new:<1,0>, old:<1,0>, ap:<255,255>, sta:<1,0>, prof:5
I (3751) wifi:state: init -> auth (b0)
Waiting for connection...
I (4751) wifi:state: auth -> init (200)
I (4761) wifi:new:<1,0>, old:<1,0>, ap:<255,255>, sta:<1,0>, prof:5
[1B][0;32mI (4761) wifi: re-trying to connect to the AP[1B][0m
Waiting for connection...
Waiting for connection...
[1B][0;32mI (6811) wifi: re-trying to connect to the AP[1B][0m
I (6821) wifi:new:<1,0>, old:<1,0>, ap:<255,255>, sta:<1,0>, prof:5
I (6831) wifi:state: init -> auth (b0)
Waiting for connection...
I (7831) wifi:state: auth -> init (200)
I (7831) wifi:new:<1,0>, old:<1,0>, ap:<255,255>, sta:<1,0>, prof:5
[1B][0;32mI (7831) wifi: re-trying to connect to the AP[1B][0m
Waiting for connection...
Waiting for connection...
[1B][0;32mI (9881) wifi: re-trying to connect to the AP[1B][0m
I (9901) wifi:new:<1,0>, old:<1,0>, ap:<255,255>, sta:<1,0>, prof:5
I (9901) wifi:state: init -> auth (b0)
Couldn't connect to the wifi. Starting WiFi AP with settings server.
[1B][0;32mI (10661) wifi: disconnecting wifi[1B][0m
I (10661) wifi:state: auth -> init (0)
I (10661) wifi:new:<1,0>, old:<1,0>, ap:<255,255>, sta:<1,0>, prof:5
I (10671) wifi:flush txq
I (10671) wifi:stop sw txq
I (10671) wifi:lmac stop hw txq
I (10671) wifi:Deinit lldesc rx mblock:10
I (10701) wifi:wifi driver task: 3ffd66b4, prio:23, stack:6656, core=0
I (10701) wifi:wifi firmware version: c7d0450
I (10701) wifi:wifi certification version: v7.0
I (10701) wifi:config NVS flash: enabled
I (10701) wifi:config nano formating: disabled
I (10711) wifi:Init data frame dynamic rx buffer num: 32
I (10711) wifi:Init management frame dynamic rx buffer num: 32
I (10721) wifi:Init management short buffer num: 32
I (10721) wifi:Init dynamic tx buffer num: 32
I (10731) wifi:Init static rx buffer size: 1600
I (10731) wifi:Init static rx buffer num: 10
I (10741) wifi:Init dynamic rx buffer num: 32
[1B][0;32mI (10741) wifi_init: rx ba win: 6[1B][0m
[1B][0;32mI (10741) wifi_init: tcpip mbox: 32[1B][0m
[1B][0;32mI (10751) wifi_init: udp mbox: 6[1B][0m
[1B][0;32mI (10751) wifi_init: tcp mbox: 6[1B][0m
[1B][0;32mI (10761) wifi_init: tcp tx win: 5744[1B][0m
[1B][0;32mI (10761) wifi_init: tcp rx win: 5744[1B][0m
[1B][0;32mI (10771) wifi_init: tcp mss: 1440[1B][0m
[1B][0;32mI (10771) wifi_init: WiFi IRAM OP enabled[1B][0m
[1B][0;32mI (10771) wifi_init: WiFi RX IRAM OP enabled[1B][0m
[1B][0;32mI (10781) wifi: initializing AP SSID: 'esp32', password: '12345678', channel: '5'[1B][0m
I (10801) wifi:mode : softAP (7c:df:a1:00:a6:1b)
I (10801) wifi:Total power save buffer number: 16
I (10801) wifi:Init max length of beacon: 752/752
I (10801) wifi:Init max length of beacon: 752/752
[1B][0;32mI (10811) settings_server: Starting settings server on port 80[1B][0m
[1B][0;32mI (10821) settings_server: Registering URI handlers[1B][0m
IP: 192.168.4.1
I (12251) wifi:new:<5,1>, old:<5,1>, ap:<5,1>, sta:<255,255>, prof:5
I (12251) wifi:station: 7c:c7:09:03:80:51 join, AID=1, bgn, 40U
[1B][0;32mI (12271) wifi: station 7c:c7:09:03:80:51 join, AID=1[1B][0m
[1B][0;32mI (12411) esp_netif_lwip: DHCP server assigned IP to a station, IP is: 192.168.4.2[1B][0m
W (14681) wifi:<ba-add>idx:4 (ifx:1, 7c:c7:09:03:80:51), tid:0, ssn:45, winSize:64
IP: 192.168.4.1
IP: 192.168.4.1
IP: 192.168.4.1
IP: 192.168.4.1
IP: 192.168.4.1
IP: 192.168.4.1
IP: 192.168.4.1
IP: 192.168.4.1
[1B][0;32mI (52351) settings_server: Accepted new settings, restarting...[1B][0m
I (54351) wifi:station: 7c:c7:09:03:80:51 leave, AID = 1, bss_flags is 658531, bss:0x3ffdef78
I (54351) wifi:new:<5,0>, old:<5,1>, ap:<5,1>, sta:<255,255>, prof:5
W (54351) wifi:<ba-del>idx
[1B][0;32mI (54351) wifi: station 7c:c7:09:03:80:51 leave, AID=1[1B][0m
I (54421) wifi:flush txq
I (54421) wifi:stop sw txq
I (54421) wifi:lmac stop hw txq
ESP-ROM:esp32s2-rc4-20191025
Build:Oct 25 2019
rst:0xc (RTC_SW_CPU_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x40024038
SPIWP:0xee
mode:DIO, clock div:1
load:0x3ffe6100,len:0x1828
load:0x4004c000,len:0x9e8
load:0x40050000,len:0x2b70
SHA-256 comparison failed:
Calculated: b4d99850fe459f149546ea9e76101766231b202d18566aeadf007958ecbb83ef
Expected: 53829e24d1b54c6c6f2f4888beb5a3b700837dc2c5a5f35b8fdc8ed825c89be5
Attempting to boot anyway...
entry 0x4004c1ec
[1B][0;32mI (42) boot: ESP-IDF v4.3-dirty 2nd stage bootloader[1B][0m
[1B][0;32mI (42) boot: compile time 15:49:06[1B][0m
[1B][0;32mI (42) boot: chip revision: 0[1B][0m
[1B][0;32mI (45) boot.esp32s2: SPI Speed      : 80MHz[1B][0m
[1B][0;32mI (50) boot.esp32s2: SPI Mode       : DIO[1B][0m
[1B][0;32mI (54) boot.esp32s2: SPI Flash Size : 4MB[1B][0m
[1B][0;32mI (59) boot: Enabling RNG early entropy source...[1B][0m
[1B][0;32mI (64) boot: Partition Table:[1B][0m
[1B][0;32mI (68) boot: ## Label            Usage          Type ST Offset   Length[1B][0m
[1B][0;32mI (75) boot:  0 nvs              WiFi data        01 02 00009000 00006000[1B][0m
[1B][0;32mI (83) boot:  1 phy_init         RF data          01 01 0000f000 00001000[1B][0m
[1B][0;32mI (90) boot:  2 factory          factory app      00 00 00010000 00100000[1B][0m
[1B][0;32mI (98) boot: End of partition table[1B][0m
[1B][0;32mI (102) esp_image: segment 0: paddr=00010020 vaddr=3f000020 size=1e688h (124552) map[1B][0m
[1B][0;32mI (137) esp_image: segment 1: paddr=0002e6b0 vaddr=3ffc6440 size=01968h (  6504) load[1B][0m
[1B][0;32mI (139) esp_image: segment 2: paddr=00030020 vaddr=40080020 size=7ebf0h (519152) map[1B][0m
[1B][0;32mI (255) esp_image: segment 3: paddr=000aec18 vaddr=3ffc7da8 size=024ech (  9452) load[1B][0m
[1B][0;32mI (257) esp_image: segment 4: paddr=000b110c vaddr=40022000 size=14438h ( 83000) load[1B][0m
[1B][0;32mI (282) esp_image: segment 5: paddr=000c554c vaddr=50000000 size=00010h (    16) load[1B][0m
[1B][0;32mI (294) boot: Loaded app from partition at offset 0x10000[1B][0m
[1B][0;32mI (294) boot: Disabling RNG early entropy source...[1B][0m
[1B][0;32mI (305) cache: Instruction cache 	: size 8KB, 4Ways, cache line size 32Byte[1B][0m
[1B][0;32mI (305) cpu_start: Pro cpu up.[1B][0m
[1B][0;32mI (319) cpu_start: Pro cpu start user code[1B][0m
[1B][0;32mI (319) cpu_start: cpu freq: 160000000[1B][0m
[1B][0;32mI (319) cpu_start: Application information:[1B][0m
[1B][0;32mI (324) cpu_start: Project name:     blink[1B][0m
[1B][0;32mI (329) cpu_start: App version:      264977f-dirty[1B][0m
[1B][0;32mI (334) cpu_start: Compile time:     Aug  4 2021 15:49:15[1B][0m
[1B][0;32mI (340) cpu_start: ELF file SHA256:  d0844d00afc260c8...[1B][0m
[1B][0;32mI (346) cpu_start: ESP-IDF:          v4.3-dirty[1B][0m
[1B][0;32mI (351) heap_init: Initializing. RAM available for dynamic allocation:[1B][0m
[1B][0;32mI (359) heap_init: At 3FF9E000 len 00002000 (8 KiB): RTCRAM[1B][0m
[1B][0;32mI (365) heap_init: At 3FFCE9B0 len 0002D650 (181 KiB): DRAM[1B][0m
[1B][0;32mI (371) heap_init: At 3FFFC000 len 00003A10 (14 KiB): DRAM[1B][0m
[1B][0;32mI (378) spi_flash: detected chip: generic[1B][0m
[1B][0;32mI (382) spi_flash: flash io: dio[1B][0m
[1B][0;32mI (390) cpu_start: Starting scheduler on PRO CPU.[1B][0m
[1B][0;32mI (414) wifi: disconnecting wifi[1B][0m
[1B][0;31mE (414) wifi_init: Failed to deinit Wi-Fi driver (0x3001)[1B][0m
[1B][0;32mI (414) wifi: connecting to wifi[1B][0m
I (424) wifi:wifi driver task: 3ffd66c4, prio:23, stack:6656, core=0
[1B][0;32mI (424) system_api: Base MAC address is not set[1B][0m
[1B][0;32mI (424) system_api: read default base MAC address from EFUSE[1B][0m
I (444) wifi:wifi firmware version: c7d0450
I (444) wifi:wifi certification version: v7.0
I (444) wifi:config NVS flash: enabled
I (444) wifi:config nano formating: disabled
I (444) wifi:Init data frame dynamic rx buffer num: 32
I (454) wifi:Init management frame dynamic rx buffer num: 32
I (454) wifi:Init management short buffer num: 32
I (464) wifi:Init dynamic tx buffer num: 32
I (464) wifi:Init static rx buffer size: 1600
I (474) wifi:Init static rx buffer num: 10
I (474) wifi:Init dynamic rx buffer num: 32
[1B][0;32mI (474) wifi_init: rx ba win: 6[1B][0m
[1B][0;32mI (484) wifi_init: tcpip mbox: 32[1B][0m
[1B][0;32mI (484) wifi_init: udp mbox: 6[1B][0m
[1B][0;32mI (494) wifi_init: tcp mbox: 6[1B][0m
[1B][0;32mI (494) wifi_init: tcp tx win: 5744[1B][0m
[1B][0;32mI (494) wifi_init: tcp rx win: 5744[1B][0m
[1B][0;32mI (504) wifi_init: tcp mss: 1440[1B][0m
[1B][0;32mI (504) wifi_init: WiFi IRAM OP enabled[1B][0m
[1B][0;32mI (514) wifi_init: WiFi RX IRAM OP enabled[1B][0m
[1B][0;32mI (524) phy_init: phy_version 1800,e7ef680,Apr 13 2021,11:45:08[1B][0m
I (684) wifi:mode : sta (7c:df:a1:00:a6:1a)
I (684) wifi:enable tsf
[1B][0;32mI (684) wifi: trying to connect to AP SSID: 'HeatPad', password: '09876543210'...[1B][0m
Waiting for connection...
I (744) wifi:new:<1,0>, old:<1,0>, ap:<255,255>, sta:<1,0>, prof:5
I (744) wifi:state: init -> auth (b0)
Waiting for connection...
I (1744) wifi:state: auth -> init (200)
I (1744) wifi:new:<1,0>, old:<1,0>, ap:<255,255>, sta:<1,0>, prof:5
[1B][0;32mI (1754) wifi: re-trying to connect to the AP[1B][0m
Waiting for connection...
Waiting for connection...
[1B][0;32mI (3804) wifi: re-trying to connect to the AP[1B][0m
I (3814) wifi:new:<1,0>, old:<1,0>, ap:<255,255>, sta:<1,0>, prof:5
I (3814) wifi:state: init -> auth (b0)
Waiting for connection...
I (4814) wifi:state: auth -> init (200)
I (4814) wifi:new:<1,0>, old:<1,0>, ap:<255,255>, sta:<1,0>, prof:5
[1B][0;32mI (4814) wifi: re-trying to connect to the AP[1B][0m
Waiting for connection...
Waiting for connection...
[1B][0;32mI (6864) wifi: re-trying to connect to the AP[1B][0m
I (6884) wifi:new:<1,0>, old:<1,0>, ap:<255,255>, sta:<1,0>, prof:5
I (6884) wifi:state: init -> auth (b0)
Waiting for connection...
I (7894) wifi:state: auth -> init (200)
I (7894) wifi:new:<1,0>, old:<1,0>, ap:<255,255>, sta:<1,0>, prof:5
[1B][0;32mI (7894) wifi: re-trying to connect to the AP[1B][0m
Waiting for connection...
Waiting for connection...
[1B][0;32mI (9944) wifi: re-trying to connect to the AP[1B][0m
I (9954) wifi:new:<1,0>, old:<1,0>, ap:<255,255>, sta:<1,0>, prof:5
I (9954) wifi:state: init -> auth (b0)
Couldn't connect to the wifi. Starting WiFi AP with settings server.
[1B][0;32mI (10684) wifi: disconnecting wifi[1B][0m
I (10684) wifi:state: auth -> init (0)
I (10684) wifi:new:<1,0>, old:<1,0>, ap:<255,255>, sta:<1,0>, prof:5
I (10694) wifi:flush txq
I (10694) wifi:stop sw txq
I (10694) wifi:lmac stop hw txq
I (10694) wifi:Deinit lldesc rx mblock:10
I (10724) wifi:wifi driver task: 3ffd66b4, prio:23, stack:6656, core=0
I (10724) wifi:wifi firmware version: c7d0450
I (10724) wifi:wifi certification version: v7.0
I (10724) wifi:config NVS flash: enabled
I (10724) wifi:config nano formating: disabled
I (10734) wifi:Init data frame dynamic rx buffer num: 32
I (10734) wifi:Init management frame dynamic rx buffer num: 32
I (10744) wifi:Init management short buffer num: 32
I (10744) wifi:Init dynamic tx buffer num: 32
I (10754) wifi:Init static rx buffer size: 1600
I (10754) wifi:Init static rx buffer num: 10
I (10764) wifi:Init dynamic rx buffer num: 32
[1B][0;32mI (10764) wifi_init: rx ba win: 6[1B][0m
[1B][0;32mI (10764) wifi_init: tcpip mbox: 32[1B][0m
[1B][0;32mI (10774) wifi_init: udp mbox: 6[1B][0m
[1B][0;32mI (10774) wifi_init: tcp mbox: 6[1B][0m
[1B][0;32mI (10784) wifi_init: tcp tx win: 5744[1B][0m
[1B][0;32mI (10784) wifi_init: tcp rx win: 5744[1B][0m
[1B][0;32mI (10794) wifi_init: tcp mss: 1440[1B][0m
[1B][0;32mI (10794) wifi_init: WiFi IRAM OP enabled[1B][0m
[1B][0;32mI (10794) wifi_init: WiFi RX IRAM OP enabled[1B][0m
[1B][0;32mI (10804) wifi: initializing AP SSID: 'esp32', password: '12345678', channel: '5'[1B][0m
I (10824) wifi:mode : softAP (7c:df:a1:00:a6:1b)
I (10824) wifi:Total power save buffer number: 16
I (10824) wifi:Init max length of beacon: 752/752
I (10824) wifi:Init max length of beacon: 752/752
[1B][0;32mI (10834) settings_server: Starting settings server on port 80[1B][0m
[1B][0;32mI (10844) settings_server: Registering URI handlers[1B][0m
IP: 192.168.4.1
I (12364) wifi:new:<5,1>, old:<5,1>, ap:<5,1>, sta:<255,255>, prof:5
I (12364) wifi:station: 7c:c7:09:03:80:51 join, AID=1, bgn, 40U
[1B][0;32mI (12424) wifi: station 7c:c7:09:03:80:51 join, AID=1[1B][0m
[1B][0;32mI (12494) esp_netif_lwip: DHCP server assigned IP to a station, IP is: 192.168.4.2[1B][0m
W (14454) wifi:<ba-add>idx:4 (ifx:1, 7c:c7:09:03:80:51), tid:0, ssn:77, winSize:64
IP: 192.168.4.1
IP: 192.168.4.1
IP: 192.168.4.1
IP: 192.168.4.1
IP: 192.168.4.1
IP: 192.168.4.1
IP: 192.168.4.1
IP: 192.168.4.1
IP: 192.168.4.1
IP: 192.168.4.1
IP: 192.168.4.1
IP: 192.168.4.1
IP: 192.168.4.1
IP: 192.168.4.1
IP: 192.168.4.1
IP: 192.168.4.1
IP: 192.168.4.1
IP: 192.168.4.1
IP: 192.168.4.1
IP: 192.168.4.1
IP: 192.168.4.1
IP: 192.168.4.1
IP: 192.168.4.1
IP: 192.168.4.1
IP: 192.168.4.1

`

**Comment by RudyFiero** [comment]
The following is the output of the ESP after I did a reset to the ESP. 

```
ESP-ROM:esp32s2-rc4-20191025
Build:Oct 25 2019
rst:0x1 (POWERON),boot:0x8 (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
mode:DIO, clock div:1
load:0x3ffe6100,len:0x1828
load:0x4004c000,len:0x9e8
load:0x40050000,len:0x2b70
SHA-256 comparison failed:
Calculated: b4d99850fe459f149546ea9e76101766231b202d18566aeadf007958ecbb83ef
Expected: 53829e24d1b54c6c6f2f4888beb5a3b700837dc2c5a5f35b8fdc8ed825c89be5
Attempting to boot anyway...
entry 0x4004c1ec
[1B][0;32mI (40) boot: ESP-IDF v4.3-dirty 2nd stage bootloader[1B][0m
[1B][0;32mI (40) boot: compile time 15:49:06[1B][0m
[1B][0;32mI (40) boot: chip revision: 0[1B][0m
[1B][0;32mI (43) boot.esp32s2: SPI Speed      : 80MHz[1B][0m
[1B][0;32mI (47) boot.esp32s2: SPI Mode       : DIO[1B][0m
[1B][0;32mI (52) boot.esp32s2: SPI Flash Size : 4MB[1B][0m
[1B][0;32mI (57) boot: Enabling RNG early entropy source...[1B][0m
[1B][0;32mI (62) boot: Partition Table:[1B][0m
[1B][0;32mI (66) boot: ## Label            Usage          Type ST Offset   Length[1B][0m
[1B][0;32mI (73) boot:  0 nvs              WiFi data        01 02 00009000 00006000[1B][0m
[1B][0;32mI (81) boot:  1 phy_init         RF data          01 01 0000f000 00001000[1B][0m
[1B][0;32mI (88) boot:  2 factory          factory app      00 00 00010000 00100000[1B][0m
[1B][0;32mI (95) boot: End of partition table[1B][0m
[1B][0;32mI (100) esp_image: segment 0: paddr=00010020 vaddr=3f000020 size=1e688h (124552) map[1B][0m
[1B][0;32mI (135) esp_image: segment 1: paddr=0002e6b0 vaddr=3ffc6440 size=01968h (  6504) load[1B][0m
[1B][0;32mI (137) esp_image: segment 2: paddr=00030020 vaddr=40080020 size=7ebf0h (519152) map[1B][0m
[1B][0;32mI (252) esp_image: segment 3: paddr=000aec18 vaddr=3ffc7da8 size=024ech (  9452) load[1B][0m
[1B][0;32mI (255) esp_image: segment 4: paddr=000b110c vaddr=40022000 size=14438h ( 83000) load[1B][0m
[1B][0;32mI (280) esp_image: segment 5: paddr=000c554c vaddr=50000000 size=00010h (    16) load[1B][0m
[1B][0;32mI (291) boot: Loaded app from partition at offset 0x10000[1B][0m
[1B][0;32mI (291) boot: Disabling RNG early entropy source...[1B][0m
[1B][0;32mI (303) ca[0F][06]žf†~[00]˜[18]†ž[1E]ÑÉÕÑ¥½¹cache 	: size 8KB, 4Ways, cache line size 32Byte[1B][0m
[1B][0;32mI (303) cpu_start: Pro cpu up.[1B][0m
[1B][0;32mI (317) cpu_start: Pro cpu start user code[1B][0m
[1B][0;32mI (317) cpu_start: cpu freq: 160000000[1B][0m
[1B][0;32mI (317) cpu_start: Application information:[1B][0m
[1B][0;32mI (322) cpu_start: Project name:     blink[1B][0m
[1B][0;32mI (326) cpu_start: App version:      264977f-dirty[1B][0m
[1B][0;32mI (332) cpu_start: Compile time:     Aug  4 2021 15:49:15[1B][0m
[1B][0;32mI (338) cpu_start: ELF file SHA256:  d0844d00afc260c8...[1B][0m
[1B][0;32mI (344) cpu_start: ESP-IDF:          v4.3-dirty[1B][0m
[1B][0;32mI (349) heap_init: Initializing. RAM available for dynamic allocation:[1B][0m
[1B][0;32mI (356) heap_init: At 3FF9E000 len 00002000 (8 KiB): RTCRAM[1B][0m
[1B][0;32mI (363) heap_init: At 3FFCE9B0 len 0002D650 (181 KiB): DRAM[1B][0m
[1B][0;32mI (369) heap_init: At 3FFFC000 len 00003A10 (14 KiB): DRAM[1B][0m
[1B][0;32mI (375) spi_flash: detected chip: generic[1B][0m
[1B][0;32mI (380) spi_flash: flash io: dio[1B][0m
[1B][0;32mI (388) cpu_start: Starting scheduler on PRO CPU.[1B][0m
[1B][0;32mI (411) wifi: disconnecting wifi[1B][0m
[1B][0;31mE (411) wifi_init: Failed to deinit Wi-Fi driver (0x3001)[1B][0m
[1B][0;32mI (411) wifi: connecting to wifi[1B][0m
I (421) wifi:wifi driver task: 3ffd66c4, prio:23, stack:6656, core=0
[1B][0;32mI (421) system_api: Base MAC address is not set[1B][0m
[1B][0;32mI (421) system_api: read default base MAC address from EFUSE[1B][0m
I (441) wifi:wifi firmware version: c7d0450
I (441) wifi:wifi certification version: v7.0
I (441) wifi:config NVS flash: enabled
I (441) wifi:config nano formating: disabled
I (441) wifi:Init data frame dynamic rx buffer num: 32
I (451) wifi:Init management frame dynamic rx buffer num: 32
I (451) wifi:Init management short buffer num: 32
I (461) wifi:Init dynamic tx buffer num: 32
I (461) wifi:Init static rx buffer size: 1600
I (471) wifi:Init static rx buffer num: 10
I (471) wifi:Init dynamic rx buffer num: 32
[1B][0;32mI (471) wifi_init: rx ba win: 6[1B][0m
[1B][0;32mI (481) wifi_init: tcpip mbox: 32[1B][0m
[1B][0;32mI (481) wifi_init: udp mbox: 6[1B][0m
[1B][0;32mI (491) wifi_init: tcp mbox: 6[1B][0m
[1B][0;32mI (491) wifi_init: tcp tx win: 5744[1B][0m
[1B][0;32mI (491) wifi_init: tcp rx win: 5744[1B][0m
[1B][0;32mI (501) wifi_init: tcp mss: 1440[1B][0m
[1B][0;32mI (501) wifi_init: WiFi IRAM OP enabled[1B][0m
[1B][0;32mI (511) wifi_init: WiFi RX IRAM OP enabled[1B][0m
[1B][0;32mI (521) phy_init: phy_version 1800,e7ef680,Apr 13 2021,11:45:08[1B][0m
I (661) wifi:mode : sta (7c:df:a1:00:a6:1a)
I (661) wifi:enable tsf
[1B][0;32mI (661) wifi: trying to connect to AP SSID: 'HeatPad', password: '09876543210'...[1B][0m
Waiting for connection...
I (671) wifi:new:<1,0>, old:<1,0>, ap:<255,255>, sta:<1,0>, prof:5
I (671) wifi:state: init -> auth (b0)
I (681) wifi:state: auth -> assoc (0)
I (691) wifi:state: assoc -> run (10)
Waiting for connection...
Waiting for connection...
I (3201) wifi:connected with HeatPad, aid = 5, channel 1, BW20, bssid = 00:17:3f:5e:36:19
I (3201) wifi:security: WPA2-PSK, phy: bg, rssi: -59
I (3211) wifi:pm start, type: 1

I (3281) wifi:AP's beacon interval = 102400 us, DTIM period = 1
Waiting for connection...
Waiting for connection...
[1B][0;32mI (4911) esp_netif_handlers: sta ip: 192.168.2.6, mask: 255.255.255.0, gw: 192.168.2.1[1B][0m
[1B][0;32mI (4911) wifi: connected[1B][0m
[1B][0;32mI (4911) wifi: got ip: 192.168.2.6[1B][0m
[1B][0;32mI (5681) settings_server: Starting settings server on port 80[1B][0m
[1B][0;32mI (5681) settings_server: Registering URI handlers[1B][0m
IP: 192.168.2.6
IP: 192.168.2.6
IP: 192.168.2.6
IP: 192.168.2.6
IP: 192.168.2.6
IP: 192.168.2.6
IP: 192.168.2.6
IP: 192.168.2.6
IP: 192.168.2.6
IP: 192.168.2.6
```

**Comment by lubosmato** [comment]
I assume this output is from example `settings_server.cpp`. 

I tried to reproduce this error but it worked properly. Can you think of a way how can I reproduce this problem?

In the first output you provided it looks like that esp32 couldn't connect to your wifi router in 10s interval. This could be caused by low wifi signal or weak power supply.

In the second output esp32 connects to your wifi properly.

*BTW Please note your wifi passwords is in logs*

**Comment by RudyFiero** [comment]
Power should be good, but I will check. Signal strength should be good. Router is in the next room. I did this at the end of the day so I didn't have time to look at it. Hopefully today I will get an answer. 

These test are from a work router that I have in my office for testing. But this reminds me of a problem I had at home. For years I had no problems sending UDP packets to some wifi outlets I have built using the ESP8266. Then one day there was a power fail. After that all (3) of my WiFi connected lights would have very poor reliability. One light I use the most was only two meters from the router. I changed position, and still poor reliability. Almost not work at all. 

The out of desperation I changed my router from channel 6 to channel 1.  After that I had prefect communications. I couldn't believe it. I don't know what the power fail had done to the router. Maybe some king of loss of calibration. I don't know.  The problem had been happening any time of the day so I didn't think it was because of other routers in the area.
